connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -282,11 +337,14 @@ public boolean startTask(
             return false;
         }
 
+        WorkerTask existing = tasks.put(id, workerTask);
```

Comment:
Yeah, we discussed this general concern briefly in last round of reviews. Originally I was thinking this wouldn't be a problem as the exception seems like it should probably be fatal for the task anyway, though I'm not sure we're currently going to clean up properly in that case anyway...